executor/common_linux.h

```diff
@@ -2514,8 +2514,12 @@ static long syz_socket_connect_nvme_tcp()
 		return -1;
 	int sock = syscall(__NR_socket, AF_INET, SOCK_STREAM, 0x0);
 	int err = errno;
-	if (setns(netns, 0))
-		fail("setns(netns) failed");
+	if (setns(netns, 0)) {
+		// The operation may fail if the fd is closed by
+		// a syscall from another thread.
+		debug("setns(netns) failed: %d\n", errno);
```

Comment:
If we fail to restore the original net namespace, we are still in init namespace. This can lead to false positive bug reports. We at least need to terminate the test process.

Reply:
Hmm, then it will be no different from `fail()` -- in case of programs that reliable close the fd, executor would just fail 10 times in a row and we'd still report a bug.

Is it actually a problem given that `syz_socket_connect_nvme_tcp()` is just a pseudo syscall and not a part of executor initialization?

Reply:
Yes, we should use exitf if this can legitimately happen.

Pity we lost of test coverage for exitf if not tested. It may start exiting all the time, and we won't notice.

I don't see difference between init/pseudo syscalls in this context.

Reply:
> I don't see difference between init/pseudo syscalls in this context.

If something fails during initialization, it's a clear sign that something is broken in the kernel or in our executor code and it's reasonable to report a problem.

Otherwise, when we're executing also other calls within the same process, everything may happen. 

> It may start exiting all the time, and we won't notice.

We might notice a coverage drop (or we may not notice, which is more likely, but it will at least be in the coverage history :) ). Just like with all other pseudo and not pseudo syscalls.

Reply:
> Otherwise, when we're executing also other calls within the same process, everything may happen.

We still try to restrict the test process from doing certain things. One of these is uncontrolled use of init net namespace. With access to init net namespace it's too easy to bring the machine down (down eth0).

Reply:
Ah, I see, here we are temporarily switching to the global networking ns and then returning back. Indeed, this pseudo syscall is special.

It would be great if we could somehow protect the fd, but I don't know how -- we have to keep it open at least for some time, so it's always vulnerable to racing operations. Adding sync primitives to ensure that no other `executor.cc` thread executes while we're in the critical section in `syz_socket_connect_nvme_tcp()` seems like an overkill to me.

I've replaced `debug()` with `exitf()` in this and in one other function, so at least the race will have to persist all 10 reexecutions, but it does not eliminate the problem completely.

Reply:
We protect kInitNetNsFd by giving it a large value.
Though, with repeat attribute programs can probably create 200 fd's. We could move it to even larger value at least on Linux.

exitf shouldn't cause failure even after 10 reexecutions, it shouldn't be reexecuted at all.

Reply:
As far as I understand, with `kInitNetNsFd` it's not that critical if syzbot mangles (probably it already does it one and off) -- all operations that depend on it would just fail at the very beginning.